crates/bevy_math/src/direction.rs

```diff
@@ -655,6 +690,15 @@ impl Dir3A {
         );
         Dir3A(quat.mul_vec3a(self.0))
     }
+
+    /// Returns `self` after an approximate normalization, assuming the value is already nearly normalized.
+    /// Useful for preventing numerical error accumulation.
```

Comment:
```suggestion
    /// Returns `self` after an approximate normalization, assuming the value is already nearly normalized.
    /// Useful for preventing numerical error accumulation.
    ///
    /// # Example
    /// The following seemingly benign code would start accumulating errors over time,
    /// leading to `my_dir` eventually not being normalized.
    /// ```rust
    /// fn system(mut my_dir: Local<Dir3>) {
    /// my_dir *= some_quaternion;
    /// method_that_assumes_the_dir_is_well_normalised(my_dir);
    /// }
    /// ```
    /// Instead, do the following.
    /// ```rust
    /// fn system(mut my_dir: Local<Dir3>) {
    /// *my_dir = (my_dir * some_quaternion).fast_renormalize();
    /// method_that_assumes_the_dir_is_well_normalised(my_dir);
    /// }
    /// ```
}
```
I liked your example from the PR :)